updates to news feed views, composables, and router links, deletion o…

trisha-alvarez opened the pull request:
- date should be automatic time stamp
   - remove date input field from new post form
- edit and delete buttons for each post on newsfeed
- button on click function for post-view.vue, use router.push
- deletepost function in composables return boolean only (for uniformity)
- rename composables files, add prefix "use",  example: use-post.ts
- formatDate should be computed()
   - switched to use-utils.ts instead
- router view link to be removed and use of router push instead in @click functions
- no console.log
- remove blank/unused post-delete.vue file

trisha-alvarez committed:
refactor: updates to news feed views, composables, and router links, deletion of post-delete file

refactor: apply code review minor revisions

fix: remove unnecessary db data

fix: incorrect router parameters

feat: vualidate for form validations

refactor: inline styling for icons

refactor: computed values for single post formattedDate variable

ci: db.json added to .gitignore file

trisha-alvarez committed:
updates to news feed views, composables, and router links, deletion of post-delete file

KarlZablan commented on db.json:
```diff
@@ -71,25 +71,18 @@
       "id": "uephXjl"
     },
     {
-      "title": "AAAA",
```
You can actually leave this blank.

KarlZablan commented on src/components/post.vue:
```diff
@@ -19,13 +31,28 @@ export default defineComponent({
-      const formatDate = (date: string): string => {
-        return moment(String(date)).format('MMMM DD, YYYY')
+    setup(props) {
+      const formattedDate = (date: string) => {
```
you can just use `props.post.date` instead of adding a parameter. Also, you can change this into a `computed` variable.

KarlZablan commented on src/components/post.vue:
```diff
@@ -19,13 +31,28 @@ export default defineComponent({
+        router.push({ 
+          name: RouteNames.POST_EDIT, 
+          params: { 
+            post: props.post.id
```
please use `postId` instead of `post`

KarlZablan commented on src/components/posts.vue:
```diff
@@ -2,19 +2,32 @@
+                    {{post.title}}
+                    <span class="text-right">
+                        <button type="button" class="btn" @click="redirectPostEdit(post.id)">
+                            <i class="fas fa-edit" style="color: #219EBC"></i>
```
avoid inline style

KarlZablan commented on src/composables/use-util.ts:
```diff
@@ -0,0 +1,9 @@
+import moment from "moment"
+
+export function formatDate(date: string): string {
```
LGTM

KarlZablan commented on src/views/news-feed.vue:
```diff
@@ -26,20 +26,20 @@ export default defineComponent({
     return {
-      newPost,
-      postsList
+      postsList,
```
you can rename this to `posts`

trisha-alvarez committed:
refactor: apply code review minor revisions

fix: remove unnecessary db data

fix: incorrect router parameters

feat: vualidate for form validations

refactor: inline styling for icons

refactor: computed values for single post formattedDate variable

trisha-alvarez committed:
ci: db.json added to .gitignore file

KarlZablan reviewed (approved):
LGTM

trisha-alvarez committed:
Merge branch '11092021-revisions' of https://github.com/trisha-alvarez/vue-newsfeed-exer-ts into 11092021-revisions
